include/diy/decomposition.hpp

```diff
@@ -314,13 +314,12 @@ decompose(int rank, const Assigner& assigner, const Creator& create)
     while (!all(offsets, 1))
     {
       // next offset
-      int i;
-      for (i = 0; i < dim; ++i)
+      for (int i = 0; i < dim; ++i)
         if (offsets[i] == 1)
           offsets[i] = -1;
         else
           break;
-      ++offsets[i];
+      ++offsets[dim];
```

Comment:
What is this? That's not equivalent to what was there before. Before it would find the first i not equal to 1 and increment it. Now it's always looking at `offsets[dim]`, which doesn't sound right at all. In fact, maybe it was a bug before that we would increment that in some cases?

Reply:
Ah, there wasn't a problem before because of `!all(offsets,1)` in the `while`-statment.

Reply:
doh! will fix. I'll just rename the `i`.

Reply:
Oh, I now understand what problem you trying to solve. Yeah, `i -> j` would be the easiest route.